fe/projects/AspNetCore/src/shared/components/Header/Header.js

```diff
@@ -1,179 +1,89 @@
-import NavigationHelper from "../../../shared/helpers/globals/NavigationHelper";
 import { Context } from "../../stores/Store";
-import { ShoppingCart } from '@mui/icons-material';
-import AuthenticationHelper from "../../../shared/helpers/globals/AuthenticationHelper";
+import ShoppingCartIcon from "../../Icons/ShoppingCart";
+import UserPopup from "../UserPopup/UserPopup";
+import SidebarMobile from "../SidebarMobile/SidebarMobile"
+import Search from "../Search/Search";
 
 function Header(props) {
     const [state, dispatch] = useContext(Context);
-    const [searchTerm, setSearchTerm] = useState(props.searchTerm ? props.searchTerm : "");
-    const [suggestions, setSuggestions] = useState([]);
     const [totalBasketItems, setTotalBasketItems] = useState(props.totalBasketItems ? props.totalBasketItems : 0);
+    const [isUserPopupOpen, setIsUserPopupOpen] = useState(false);
+    const [isLanguageSwitechOpen, setIsLanguageSwitechOpen] = useState(false);
+    const overlayRef = useRef(null);
 
-    const getSuggestionValue = (suggestion) => {
-        return suggestion;
-    };
-
-    const renderSuggestion = (suggestion) => {
-        return (
-            <div className="suggestion">
-                {suggestion}
-            </div>
-        );
-    };
-
-    const onSuggestionsFetchRequested = (args) => {
-        setSearchTerm(args.value);
-
-        if (args.value && args.value.length >= HeaderConstants.minSearchTermLength()) {
-        
-            dispatch({ type: "SET_IS_LOADING", payload: true });
-
-            const searchParameters = {
-
-                searchTerm: args.value,
-                size: HeaderConstants.searchSuggenstionsSize()
-            };
-
-            const requestOptions = {
-                method: "GET",
-                headers: { "Content-Type": "application/json", "X-Requested-With": "XMLHttpRequest" }
-            };
-
-            const url = props.getSuggestionsUrl + "?" + QueryStringSerializer.serialize(searchParameters);
-
-            return fetch(url, requestOptions)
-            .then(function (response) {
-                dispatch({ type: "SET_IS_LOADING", payload: false });
-
-                AuthenticationHelper.HandleResponse(response);
-
-                return response.json().then(jsonResponse => {
-
-                    if (response.ok) {
-
-                        setSuggestions(() => []);
-                        setSuggestions(() => jsonResponse);
-                    }
-                    else {
-                        toast.error(props.generalErrorMessage);
-                    }
-                });
-            }).catch(() => {
-                dispatch({ type: "SET_IS_LOADING", payload: false });
-                toast.error(props.generalErrorMessage);
-            });
+    useEffect(() => {
+        const totalItems = state.totalBasketItems;
+        if (totalItems != null && totalItems < props.totalBasketItems) {
+            state.totalBasketItems = props.totalBasketItems;
```

Comment:
**suggestion:** manipulating state directly, especially within useEffect is a bad idea, let's change it - please check in chat gpt how to do it